core/src/main/java/com/aem/demo/core/models/NewsCardModel.java

```diff
@@ -62,44 +68,35 @@ public String getFullPath() {
     }
 
 
-    public int  getLikes() {
-        int likes = 0;
 
+    @PostConstruct
+    private void init() {
         String rootPath = resource.getPath();
-
-        resource = resourceResolver.getResource(rootPath);
-        ModifiableValueMap properties = resource != null ? resource.adaptTo(ModifiableValueMap.class) : null;
-
-        if (properties != null) {
-            likes = resource.adaptTo(ValueMap.class).get("likes", 0);
-            properties.put("likes", likes + 1);
-        }
-        try {
-            resourceResolver.commit();
-        } catch (PersistenceException e) {
-            log.info(e.getMessage());
+        String dislikeParam = servletRequest.getParameter("dislike");
+        String likeParam = servletRequest.getParameter("like");
+        if (likeParam != null) {
+            likes = incrementLikeDislike(likeParam, rootPath);
+        } else if (dislikeParam != null) {
+            dislikes = incrementLikeDislike(dislikeParam, rootPath);
         }
-        return likes;
-    }
 
-    public int  getDisLikes() {
-        String dislikeParameter = servletRequest.getParameter("dislikeId");
-        int dislikes = 0;
+    }
 
-        String rootPath = resource.getPath();
 
+    private int incrementLikeDislike(String property, String rootPath) {
+        int incrementProperty = 0;
         resource = resourceResolver.getResource(rootPath);
```

Comment:
Still a good old joke)
You have the `resource` field injected in line 30,
You are apparently sure this field is not null, because in line 74 you are retrieving the path form this resource via `String rootPath = resource.getPath();` without any nullity check.
But okay.
Then, in line 88, you use `resourceResolver` to get a resource by the path that you retrieved from the resource in line 74.
And then you assign the result back to thre `resource` field)

Let me re-iterate.
You use resource. To get the path from it. To find a resource with the very path you got from the very resource. And then assign it back to the very resource!))) 
And then you check the resource for nullity. You check for nullity the very resource that you already used to get the path that you used to get back the very resource!))

Is this some kind of prank?) 

It is as if you yourself stood in the middle of a room calling out "Gaybulokhon Musabaev! Is Gaybulokhon Musabaev here!?" And then listened carefully if someone else replies to the call.